File: app/src/main/java/com/imperium/academio/ui/adapters/MaterialTopicRvAdapter.java
package com.imperium.academio.ui.adapters;

import android.app.Activity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.imperium.academio.R;
import com.imperium.academio.ui.adapters.helper.TopicRvViewHolder;
import com.imperium.academio.ui.model.MaterialTopicRvModel;

import java.util.List;


public class MaterialTopicRvAdapter extends RecyclerView.Adapter<TopicRvViewHolder> {

    final Activity activity;
    final List<MaterialTopicRvModel> items;
    private OnTopicItemClickListener listener;
    int selectedTopicIndex = -1;

    public MaterialTopicRvAdapter(Activity activity, List<MaterialTopicRvModel> items) {
        this.activity = activity;
        this.items = items;
    }

    public void setOnTopicItemClickListener(OnTopicItemClickListener listener) {
        this.listener = listener;
    }

    @NonNull
    @Override
    public TopicRvViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType ) {
        View view = LayoutInflater.from(activity).inflate(R.layout.template_material_topic, parent, false);
        return new TopicRvViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull TopicRvViewHolder viewHolder, int position) {
        MaterialTopicRvModel item = items.get(position);
        viewHolder.topic.setText(item.getTopic());

        viewHolder.linearLayout.setOnClickListener(v -> {
            selectedTopicIndex = selectedTopicIndex == position ? -1 : position;
            listener.onTopicItemClick(viewHolder.itemView, position);
        });

        viewHolder.linearLayout.setBackgroundResource(
                selectedTopicIndex == position ?
                        R.drawable.material_rvi_selected_bg :
                        R.drawable.material_rvi_bg
        );
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public interface OnTopicItemClickListener {
        void onTopicItemClick(View itemView, int position);
    }
}

File: app/src/main/java/com/imperium/academio/ui/adapters/MaterialTopicRvAdapter.kt
package com.imperium.academio.ui.adapters

import android.app.Activity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.imperium.academio.R
import com.imperium.academio.ui.adapters.helper.TopicRvViewHolder
import com.imperium.academio.ui.model.MaterialTopicRvModel

class MaterialTopicRvAdapter(val activity: Activity, private val items: List<MaterialTopicRvModel>) : RecyclerView.Adapter<TopicRvViewHolder>() {
    private var listener: OnTopicItemClickListener? = null
    private var selectedTopicIndex = -1
    fun setOnTopicItemClickListener(listener: OnTopicItemClickListener?) {
        this.listener = listener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TopicRvViewHolder {
        val view = LayoutInflater.from(activity).inflate(R.layout.template_material_topic, parent, false)
        return TopicRvViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: TopicRvViewHolder, position: Int) {
        val item = items[position]
        viewHolder.topic.text = item.topic
        viewHolder.linearLayout.setOnClickListener {
            selectedTopicIndex = if (selectedTopicIndex == position) -1 else position
            listener!!.onTopicItemClick(viewHolder.itemView, position)
        }
        viewHolder.linearLayout.setBackgroundResource(
                if (selectedTopicIndex == position) R.drawable.material_rvi_selected_bg else R.drawable.material_rvi_bg
        )
    }

    override fun getItemCount(): Int {
        return items.size
    }

    interface OnTopicItemClickListener {
        fun onTopicItemClick(itemView: View?, position: Int)
    }
}